Tests/SentryTests/SentryHubTests.swift

```diff
@@ -371,21 +371,53 @@ class SentryHubTests: XCTestCase {
         }
     }
     
-    func testStartTransaction_WhenProfilingDisabled_DoesNotCaptureProfile() {
-        let options = fixture.options
-        options.enableProfiling = false
-        options.tracesSampler = {(_: SamplingContext) -> NSNumber in
-            return 1
+    func testStartTransaction_NotSamplingProfileUsingEnableProfiling() {
+        testProfilesSampler(expected: .no) { options in
+            options.enableProfiling_DEPRECATED_TEST_ONLY = false
         }
-        let hub = fixture.getSut(options)
-        let span = hub.startTransaction(name: fixture.transactionName, operation: fixture.transactionOperation)
-        span.finish()
-        
-        guard let additionalEnvelopeItems = fixture.client.captureEventWithScopeInvocations.first?.additionalEnvelopeItems else {
-            XCTFail("Expected to capture at least 1 event")
-            return
+    }
+    
+    func testStartTransaction_SamplingProfileUsingEnableProfiling() {
+        testProfilesSampler(expected: .yes) { options in
+            options.enableProfiling_DEPRECATED_TEST_ONLY = true
+        }
+    }
+    
+    func testStartTransaction_NotSamplingProfileUsingSampleRate() {
+        testProfilesSampler(expected: .no) { options in
+            options.profilesSampleRate = 0.49
+        }
+    }
+    
+    func testStartTransaction_SamplingProfileUsingSampleRate() {
+        testProfilesSampler(expected: .yes) { options in
+            options.profilesSampleRate = 0.5
+        }
+    }
+    
+    func testStartTransaction_SamplingProfileUsingProfilesSampler() {
+        testProfilesSampler(expected: .yes) { options in
+            options.profilesSampler = { _ in return 0.51 }
```

Comment:
I don't understand how some of these tests work. With a 0.51 sample rate, this should fail 49% of the time. Is something else getting mocked to make this always work? Same for a few others with a rate that's not 0.0 or 1.0.

Reply:
The tests don't use a random value. The random value is fixed see https://github.com/getsentry/sentry-cocoa/blob/45ed684331da1b094637dcc6de34a833e0e19d6d/Tests/SentryTests/SentryHubTests.swift#L779-L781

Reply:
OK, after reading SentryProfilesSampler's implementation I see how this works now, thanks.